### DISCUSSION TITLE
Looking for Examples

### DISCUSSION THREAD

**Discussion by MikeBAlgWare** [olsh/todoist-net#49] [Q&A] [answered]
Hi, loving this library so far, it's really helped me get over some hurdles in a hurry.  Is there any more documentation or examples available?  

Specifically I'd like to know how to get a single task by id.  Is there a method to do this besides pulling the entire collection down and checking each object?  

Also, I am having trouble updating DueDate in the Task object- I am able to initially set it in the quick add, but this will need to be updated in existing tasks from time to time..    Thanks, great job on this- Mike

**Comment by AhmedZaki99** [comment, marked as answer]
Hi Mike,
IntelliSense in Visual Studio and VS Code helped me a lot when I was discovering this library for the first time. You will find that the [`TodoistClient`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/TodoistClient.cs) class is well commented and it has a structure very similar to Todoist API itself.

For example, in Todoist docs, you can find helpful information about how to get a single task when you go to [this section](https://developer.todoist.com/sync/v9/#get-item-info): 
Sync API -> Items -> Get item info

And in `TodoistClient`, you can get a single task using this method:
`TodoistClient.Items.GetAsync(ComplexId, CancellationToken)`

That's why after reading the [Todoist Sync API docs](https://developer.todoist.com/sync/v9), IntelliSense will be your guide:

![image](https://github.com/olsh/todoist-net/assets/44468352/f9fa0ba1-a879-4635-8eef-91d311a95512)

Similarly, you can find a method to update tasks (including due dates) if you navigate a little in IntelliSense's menu:

![image](https://github.com/olsh/todoist-net/assets/44468352/9eb1ca9b-6020-46d4-86e4-fd9a239f4f3c)

And again, this method is just a wrapper for an endpoint in Todoist API, which you can find documented in the [update-an-item](https://developer.todoist.com/sync/v9/#update-an-item) section.

**Reply by MikeBAlgWare** [reply]
Hi, thanks for the response and tips.  I'm OK on getting an item now, but I'm stuck on updating the due date.  I am assuming that task.DueDate is looking for a DueDate object and not just a DateTime.  I'm not understanding how to build a DueDate...all the properties I have tried to set are read only.

Whoops- never mind, I figured it out.  I needed to build a DueDate with DueDate. FromText() or DueDate.CreateFullDay()

**Reply by AhmedZaki99** [reply]
You might already know that due dates in Todoist have three different types, as documented [here](https://developer.todoist.com/sync/v9/#full-day-dates):

> Todoist supports three types of due dates.
> 
> - Full-day dates (like "1 January 2018" or "tomorrow")
> - Floating due dates with time (like "1 January 2018 at 12:00" or "tomorrow at 10am")
> - Due dates with time and fixed timezone (like "1 January 2018 at 12:00 America/Chicago" or "tomorrow at 10am Asia/Jakarta")
>
> Unless specified explicitly, dates with time are created as floating.

That's why when you want to create a `DueDate` object, you have four available static methods, one for each type, and one to create dates from text:

- [`CreateFullDay(DateTime date)`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Models/DueDate.cs#L144)
- [`CreateFloating(DateTime dateTime)`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Models/DueDate.cs#L169)
- [`CreateFixedTimeZone(DateTime dateTime, string timezone)`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Models/DueDate.cs#L190)
- [`FromText(string text, Language language = null)`](https://github.com/olsh/todoist-net/blob/master/src/Todoist.Net/Models/DueDate.cs#L125)

And you can use them like this:

```c#
// Full Day
var item = new Item("Full day task") { DueDate = DueDate.CreateFullDay(new DateTime(2021, 10, 22)) };

// Floating
var item = new Item("Floating date task") { DueDate = DueDate.CreateFloating(DateTime.Now) };

// From Text
var item = new Item("Text date task") { DueDate = DueDate.FromText("tomorrow", Language.English) };
```

**Reply by MikeBAlgWare** [reply]
Awesome, thank you so much.  I am learning a lot- longtime ToDoist user, longtime coder, rookie web API coder...this library is really great
